### DISCUSSION TITLE
What about selective routing for the netflex addon to aviod VPN

### DISCUSSION THREAD

**Discussion by requa3r0** [CastagnaIT/plugin.video.netflix#1152] [Ideas]
Hi
Most run a vpn on their system. 
This often gives problems with Netflex and therfore also this addon.

I see a need to do some sort of selective routing of the Netflex trafix to the WAN interface, rather the VPN interface. To avoid the VPN...not the other way around to try to geo-hack netflex. There are other ways to do this if this is your purpose ;O)

Regardless if your vpn connection is done running a app that creates a vpn interface, or if you run a  proper vpn server, and connect through a gateway, I think it would be great if the netflex addon cold be set to bind to a specific interface or gateway!

This way all of the media center internet traffic would be through the vpn as per the standard OS setting, but the netflex addon traffic could then be forced through the WAN ensuring propper compliance with netflex (stupid vpn regulations).

note:
If im connected to vpn when running the Netflex addon, it seems to work, but it lags and stops and half the content (newer episodes etc) seems to be inaccessible.

Its quite strange.

Is there a way to fix this.

Doing selective routing in IP-tables or with policies etc to route all netflex traffic seems to be a lost course.
According to various post, you effectually need to whitelist half the internet to make it work ;O/

Requaero.

**Comment by CastagnaIT** [comment]
this seems very peculiar to me, in theory there should be a somewhat hacked possibility to change source interface ip,
but i don't know it is feasible with the Request module used by the addon
and there are also other complications:
the "Request" module will soon have to be replaced with an http2 module (so even if there might be a solution it will be broken)
all https requests for audio/video/subtitle streams are not handled by the addon but from InputStreamAdaptive there is no way to have some type of control
this hack need to work also between all types of operating systems that run Kodi then need to be tested on all operative systems

I have read on net that some VPN software provides ways to exclude web addresses, i think this would be the cleanest solution

**Comment by CastagnaIT** [comment]
> If im connected to vpn when running the Netflex addon, it seems to work, but it lags and stops and half the content (newer episodes etc) seems to be inaccessible.
> 
> Its quite strange.

netflix makes several geographical checks not only for GEO lock purpose, but are needed to choose their server connections
so with VPN it is possible that the wrong servers are selected, causing problems with the API's
I don't worry about that...

**Reply by requa3r0** [reply]
My selected vpn server is the closest one, im my own country and home town.
Im not using vpn to change geo location. 

But none the less...netflex detects it and makes everything quite sketchy

Its not trivial to whitelist netflex...there are an endless storm of servers and subdomains.
